components/brave_vpn/browser/brave_vpn_service.cc

```diff
@@ -716,28 +745,28 @@ void BraveVpnService::OnGetSubscriberCredentialV12(
       return;
     }
 
-    // If we get here, we've already tried two credentials (the retry failed).
+    // We can set the state as FAILED and do not attempt to get another
+    // credential. The cached credential will eventually expire and user will
+    // fetch a new one.
+    //
+    // There could be two reasons for this.
+
+    // 1. We've already tried two credentials (the retry failed).
     if (token_no_longer_valid && IsRetriedSkusCredential(local_prefs_)) {
       VLOG(2) << __func__
               << " : Got TokenNoLongerValid again with retried skus credential";
+      out_of_credentials_ = true;
```

Comment:
we need this setting here?

Reply:
No, I don't think we do - good catch 😄 The conditions for `UpdatePurchasedStateForSessionExpired` should already match - I can remove this!

Reply:
Fixed!